### PR TITLE
Basic UI styling

### PR DESCRIPTION
**Pull request by ifeins**
What's not included:
- Styling for Onboarding Flow
- Styling for request states - pending accept, accepted, declined.
- Styling for empty states: no requests, no shoppers, no items in unified shopping list, etc.
## Home screen

home
## Trip flow

pending-requests

request-details

unified-shopping-list
## Request flow:

select-shopper

create-request

### CODE REVIEW

File: app/src/main/java/com/iic/shopingo/ui/home/ActionCardView.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      if (resourceId != -1) {
+        icon.setImageResource(resourceId);
+      }
+    } finally {
+      typedArray.recycle();
+    }
+  }
+
+  @Override
+  protected void onLayout(boolean changed, int l, int t, int r, int b) {
```
Comment (assafgelber) [inline app/src/main/java/com/iic/shopingo/ui/home/ActionCardView.java]: What are these?

File: app/src/main/java/com/iic/shopingo/ui/home/ActionCardView.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    int[] position = new int[2];
+    actionButton.getLocationOnScreen(position);
+  }
+
+  public void setListener(Listener listener) {
+    this.listener = listener;
+  }
+
+  @OnClick(R.id.action_card_button)
+  public void onClick(View target) {
```
Comment (assafgelber) [inline app/src/main/java/com/iic/shopingo/ui/home/ActionCardView.java:80]: Try to see if you can use google's cardview thing with a proper footer button, or whatever they call it. Could also be useful for other listviews if we want them to be cards as well.

File: app/src/main/java/com/iic/shopingo/ui/request_flow/activities/CreateShoppingListActivity.java [hunk 3]
```diff
@@ -49,17 +48,38 @@ protected void onCreate(Bundle savedInstanceState) {
...
+  @Override
+  public void onTextChanged(CharSequence s, int start, int before, int count) {
+  }
+
+  @Override
+  public void afterTextChanged(Editable s) {
+    String symbol = Currency.getInstance(Locale.getDefault()).getSymbol();
+
+    String text = s.toString();
+    if (!text.contains("$")) {
```
Comment (assafgelber) [inline app/src/main/java/com/iic/shopingo/ui/request_flow/activities/CreateShoppingListActivity.java:74]: There has to be a better way to do this. This is really weird.

Comment (assafgelber) [inline app/src/main/java/com/iic/shopingo/ui/request_flow/activities/CreateShoppingListActivity.java:74]: And either way: `!text.contains(symbol)`

File: app/src/main/java/com/iic/shopingo/ui/trip_flow/activities/ManageTripActivity.java [hunk 4]
```diff
@@ -119,4 +151,22 @@ public void onRequestSelected(IncomingRequest request) {
     pager.setCurrentItem(1, true);
     // TODO: scroll to correct shopping list
   }
+
+  @Override
+  public void onDiscardTripDialogOK() {
```
Comment (assafgelber) [inline app/src/main/java/com/iic/shopingo/ui/trip_flow/activities/ManageTripActivity.java:156]: @AsafGartner `// TODO: Add api call to end trip`

Comment (assafgelber) [inline app/src/main/java/com/iic/shopingo/ui/trip_flow/activities/ManageTripActivity.java:156]: BTW, what happens if you cancel the trip with accepted requests?

File: app/src/main/java/com/iic/shopingo/ui/trip_flow/views/RequestListItem.java [hunk 5]
```diff
@@ -41,9 +43,13 @@ public RequestListItem(Context context, AttributeSet attrs, int defStyleAttr) {
...
     if (thumbnail != null) {
       this.thumbnail.setImageBitmap(thumbnail);
+    } else {
+      // TODO: remove this code
+      Picasso.with(getContext()).load(AvatarUriGenerator.generateAvatarUri(name)).into(this.thumbnail);
     }
+
     this.name.setText(name);
-    this.numItems.setText(Integer.toString(numItems));
+    this.numItems.setText(String.format("%d items", numItems));
```
Comment (assafgelber) [inline app/src/main/java/com/iic/shopingo/ui/trip_flow/views/RequestListItem.java]: Move to strings, along with pluralization stuff.

File: app/src/main/res/layout/activity_create_request.xml [hunk 6]
```diff
@@ -4,42 +4,73 @@
     tools:context="com.iic.shopingo.ui.request_flow.activities.CreateShoppingListActivity"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
+    android:fillViewport="true"
     >
   <LinearLayout
       android:layout_width="match_parent"
       android:layout_height="match_parent"
       android:orientation="vertical"
+      android:paddingTop="16dp"
```
Comment (assafgelber) [inline app/src/main/res/layout/activity_create_request.xml:13]: Move the numbers to dimens

File: app/src/main/res/layout/activity_home.xml [hunk 7]
```diff
@@ -7,22 +8,34 @@
     android:paddingTop="@dimen/activity_vertical_margin"
     android:paddingBottom="@dimen/activity_vertical_margin"
     tools:context="com.iic.shopingo.ui.HomeActivity"
-    android:gravity="center_vertical|center_horizontal"
-    android:orientation="vertical"
-    >
+    android:gravity="center_horizontal|center_vertical"
```
Comment (assafgelber) [inline app/src/main/res/layout/activity_home.xml]: I think you can just do `center` instead, no?

File: app/src/main/res/layout/activity_home.xml [hunk 8]
```diff
@@ -7,22 +8,34 @@
...
+      android:orientation="vertical"
+      android:layout_width="match_parent"
       android:layout_height="wrap_content"
-      android:text="@string/home_create_request_btn_title"
-      android:id="@+id/home_create_request_btn"
-      android:layout_gravity="center_horizontal"
+      android:layout_marginTop="16dp"
+      android:background="@drawable/action_card_background"
+      android:padding="16dp"
+      android:elevation="2dp"
```
Comment (assafgelber) [inline app/src/main/res/layout/activity_home.xml]: Move these to a `style` if you're gonna reuse them.

File: app/src/main/res/layout/activity_request_details.xml [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+        android:layout_width="75dp"
+        android:layout_height="75dp"
+        />
+    <TextView
+        android:id="@+id/request_details_requester_name"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="center_vertical"
+        android:layout_marginLeft="8dp"
+        android:text="Moshe"
```
Comment (assafgelber) [inline app/src/main/res/layout/activity_request_details.xml:28]: If we use placeholders like this, we need to make sure there is absolutely no way we don't replace these with actual content somehow.

File: app/src/main/res/values/strings_request_flow.xml [hunk 10]
```diff
@@ -20,4 +20,7 @@
   <string name="request_state_settle_button">Done!</string>
   <string name="request_state_try_again_button">Try someone else</string>
   <string name="request_state_go_yourself_button">Go yourself</string>
+  
+  <string name="format_offered_price">Offered price: <xliff:g id="price">%1$s</xliff:g></string>
```
Comment (assafgelber) [inline app/src/main/res/values/strings_request_flow.xml]: `%.2f`

File: app/src/main/java/com/iic/shopingo/ui/request_flow/activities/CreateShoppingListActivity.java (hunk 3 above)
Comment (assafgelber) [inline app/src/main/java/com/iic/shopingo/ui/request_flow/activities/CreateShoppingListActivity.java:74]: Also, we don't really handle different currencies in the server, so might as well just leave it as `$` for now.

File: app/src/main/java/com/iic/shopingo/ui/home/ActionCardView.java (hunk 1 above)
Comment (ifeins) [inline app/src/main/java/com/iic/shopingo/ui/home/ActionCardView.java]: It's a method I've used for debugging, I'll remove it.

**Pull request merged**